build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/ElasticsearchTestBasePlugin.java

```diff
@@ -108,7 +108,10 @@ public void execute(Task t) {
                 "--add-opens=java.base/java.nio.file=ALL-UNNAMED",
                 "--add-opens=java.base/java.time=ALL-UNNAMED",
                 "--add-opens=java.management/java.lang.management=ALL-UNNAMED",
-                "-XX:+HeapDumpOnOutOfMemoryError"
+                "-XX:+HeapDumpOnOutOfMemoryError",
+                // REMOVE once bumped to a JDK greater than 21.0.1, https://github.com/elastic/elasticsearch/issues/103004
```

Comment:
I'm not sure we'll be able to remove this just because of a JDK upgrade since we still test on older JDKs, unless this fix is backported to JDK 17.

Reply:
That's a fair comment @mark-vieira.   We're currently in the process of investigating a possible "fix" in Lucene. If this were to happen then these options can be removed. If not, then I'll amend the comment to something more appropriate for its longer term presence.